tsconfig.json

```diff
@@ -7,7 +7,7 @@
     "outDir": "lib",
     "rootDir": "src",
     "strict": true,
-    "target": "es2017"
+    "target": "es2019"
```

Comment:
This is the riskiest part of this PR. I don't really know what the consequences are of a change like this. I needed it to use `Array.flat` -- if this change seems like a bad idea, I can pull in something from `lodash` instead.

Reply:
Based on [this StackOverflow answer](https://stackoverflow.com/a/59787575), it looks like it depends on which NodeJS engine we're stating compatibility with. Looks like we're currently specifying `>= 8.0.0` ([source](https://github.com/artsy/cli/blob/ada44219a4b971f7391613def51c341aedc81d1f/package.json#L60-L62)), but I think we should be able to bump that to `>= 12.0.0` which is 100% compatible with ES2019.

If we wanted to use some ES2020 features, we could either bump that engine requirement to `> 12.9.0` and include `"lib": ["es2020"]` or update the target to `es2020` and up the engine requirement to `>=14.0.0`.

Reply:
Went ahead and made that package.json engine update!